core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -135,6 +134,7 @@
 import static java.util.Collections.emptyMap;
 import static org.elasticsearch.index.engine.Engine.Operation.Origin.PRIMARY;
 import static org.elasticsearch.index.engine.Engine.Operation.Origin.REPLICA;
+import static org.hamcrest.CoreMatchers.instanceOf;
```

Comment:
I added unit test for `checkIfDocumentFailureOrThrow` and test mocking non-tragic exception from index writer to test failure handling when executing write/delete operations.